test/Main.hs

```diff
@@ -0,0 +1,208 @@
+
+-- | Discover all .agda files under the Fail directory.
+discoverFailTests :: FilePath -> FilePath -> IO [TestTree]
+discoverFailTests testDir buildDir = do
+  agdaFiles <- findAgdaFilesRecursive (testDir </> "Fail")
+  forM (sort agdaFiles) $ \agdaFile -> do
+    let testName = dropExtension (makeRelative (testDir </> "Fail") agdaFile)
+        goldenFile = dropExtension agdaFile ++ ".err"
+    return $ failTest testDir buildDir testName agdaFile goldenFile
+
+-- | Find all .agda files recursively in a directory.
+findAgdaFilesRecursive :: FilePath -> IO [FilePath]
+findAgdaFilesRecursive dir = do
+  contents <- listDirectory dir
+  paths <- forM contents $ \name -> do
+    let path = dir </> name
+    isDir <- doesDirectoryExist path
+    if isDir
+      then findAgdaFilesRecursive path
+      else return [path | ".agda" `isSuffixOf` name]
+  return (concat paths)
+
+-- | Check if path is a directory.
+doesDirectoryExist :: FilePath -> IO Bool
+doesDirectoryExist path = do
+  exists <- doesFileExist path
+  if exists
+    then return False
+    else do
+      -- If it's not a file, check if it's a directory
+      contents <- listDirectory path `catch` \(_ :: SomeException) -> return []
+      return (not (null contents) || path `elem` ["Succeed", "Fail", "Cubical"])
+
+-- | Create a test for a succeed case.
+-- Runs agda2hs on the .agda file, compares the output .hs to the golden file,
+-- then compiles the .hs file with ghc.
+succeedTest :: FilePath -> FilePath -> String -> FilePath -> FilePath -> TestTree
+succeedTest testDir buildDir testName agdaFile goldenFile =
+  goldenVsStringDiff testName diffCmd goldenFile $ do
+    let succeedDir = testDir </> "Succeed"
+        -- Output to build directory to avoid polluting source tree
+        outDir = buildDir </> "Succeed"
+        -- Get the relative path from Succeed to the agda file
+        relAgdaPath = makeRelative succeedDir agdaFile
+        -- Compute the expected output file path (same relative path, but .hs)
+        generatedFile = outDir </> dropExtension relAgdaPath ++ ".hs"
+        generatedDir = takeDirectory generatedFile
+
+    -- Ensure output directory exists (including subdirectories)
+    createDirectoryIfMissing True generatedDir
+
+    -- Run agda2hs with include path for Succeed directory
+    (exitCode, stdout, stderr) <- readProcessWithExitCode
+      "agda2hs"
+      [agdaFile, "-o", outDir, "-v0", "-i" ++ succeedDir]
+      ""
+
+    case exitCode of
+      ExitSuccess -> do
+        -- Read the generated .hs file from build directory
+        content <- LBS.readFile generatedFile
+
+        -- Also run ghc to check the generated code compiles
+        -- Add include path for finding imported modules
+        (ghcExit, ghcOut, ghcErr) <- readProcessWithExitCode
+          "ghc"
+          ["-fno-code", "-i" ++ outDir, generatedFile]
+          ""
+
+        case ghcExit of
+          ExitSuccess -> return content
```

Comment:
Inline `content` here and comment that the generated `.hs` file is the golden value returned here.